File: 0260.SingleNumberIII.go
func singleNumber(nums []int) []int {
	diff := 0
	for _, num := range nums {
		diff ^= num
	}
	diff &= -diff // get highest bit
	ans := []int{0, 0}
	for _, num := range nums {
		if num&diff != 0 {
			ans[0] ^= num
		} else {
			ans[1] ^= num
		}
	}
	return ans
}

File: 0260.SingleNumberIII.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        t = ab = reduce(operator.xor, nums)
        t &= -t
        a = reduce(operator.xor, filter(lambda x: x&t, nums))
        return [a, a^ab]
        '''
        ab = reduce(operator.xor, nums)
        tmp = 1
        while tmp&ab == 0:
            tmp <<= 1
        nums1 = []
        for num in nums:
            if num & tmp:
                nums1.append(num)
        a = reduce(operator.xor, nums1)
        return [a, a^ab]
        '''